File: Sorting Algorithms/Radix Sort/RadixSort.java
public class RadixSort {
    
    // A utility function to get the maximum value in an array
    static int getMax(int arr[], int n) {
        int mx = arr[0];
        for (int i = 1; i < n; i++)
            if (arr[i] > mx)
                mx = arr[i];
        return mx;
    }
    
    // A function to do counting sort of arr[] according to the digit represented by exp.
    static void countSort(int arr[], int n, int exp) {
        int output[] = new int[n]; // output array
        int i;
        int count[] = new int[10];
        Arrays.fill(count, 0);
        
        // Store count of occurrences in count[]
        for (i = 0; i < n; i++)
            count[(arr[i] / exp) % 10]++;
        
        // Change count[i] so that count[i] now contains actual position of this digit in output[]
        for (i = 1; i < 10; i++)
            count[i] += count[i - 1];
        
        // Build the output array
        for (i = n - 1; i >= 0; i--) {
            output[count[(arr[i] / exp) % 10] - 1] = arr[i];
            count[(arr[i] / exp) % 10]--;
        }
        
        // Copy the output array to arr[], so that arr[] now contains sorted numbers according to the current digit
        for (i = 0; i < n; i++)
            arr[i] = output[i];
    }
    
    // The main function to implement Radix Sort
    static void radixSort(int arr[], int n) {
        int m = getMax(arr, n);
        
        // Do counting sort for every digit. Note that instead of passing digit number, exp is passed. exp is 10^i where i is the current digit number
        for (int exp = 1; m / exp > 0; exp *= 10)
            countSort(arr, n, exp);
    }
    
    // A utility function to print an array
    static void printArray(int arr[], int n) {
        for (int i = 0; i < n; i++)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
    
    // Driver code to test above
    public static void main(String args[]) {
        int arr[] = {170, 45, 75, 90, 802, 24, 2, 66};
        int n = arr.length;
        radixSort(arr, n);
        printArray(arr, n);
    }
}

File: Sorting Algorithms/Radix Sort/RadixSort.cs
using System;

class RadixSort
{
    public static void Radix_Sort(int[] arr)
    {
        int i, j;
        int[] tmp = new int[arr.Length];
        int[] count = new int[256];

        for (int shift = 0; shift < 32; shift += 8)
        {
            count = new int[256];
            int mask = 255 << shift;

            foreach (int num in arr)
            {
                count[(num & mask) >> shift]++;
            }

            for (i = 1; i < 256; i++)
            {
                count[i] += count[i - 1];
            }

            for (j = arr.Length - 1; j >= 0; j--)
            {
                tmp[--count[(arr[j] & mask) >> shift]] = arr[j];
            }

            tmp.CopyTo(arr, 0);
        }
    }

    static void Main()
    {
        int[] arr = { 170, 45, 75, 90, 802, 24, 2, 66 };
        Radix_Sort(arr);

        Console.WriteLine("Sorted array:");
        foreach (int num in arr)
        {
            Console.Write(num + " ");
        }
    }
}